emukit/quadrature/acquisitions/uncertainty_sampling.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+import numpy as np
+from typing import Tuple, Union
+
+from ...core.acquisition import Acquisition
+from ...core.interfaces import IDifferentiable
+from ...quadrature.methods import WarpedBayesianQuadratureModel
+
+
+class UncertaintySampling(Acquisition):
+    """
+    Uncertainty sampling acquisition function for (warped) Bayesian quadrature.
+
+    The variance of the approximate transformed GP is used. If the integration measure is a probability measure,
+    then the variance will we weighted with the probability density at each point.
+    """
+
+    def __init__(self, model: Union[WarpedBayesianQuadratureModel, IDifferentiable]):
+        """
+        :param model: A warped Bayesian quadrature model that has predictive gradients
+        """
+        self.model = model
+
+    def has_gradients(self) -> bool:
+        return True
+
+    def _evaluate(self, x: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Evaluates the acquisition function. If a probability measure is used, then the variance will be weighted by
+        the density of the probability measure.
+
+        :param x: (n_points x n_dims) array of points at which to calculate acquisition function values
+        :return: (n_points x 1) array of acquisition function values, unweighted variances
+        """
+        variances = self.model.predict(x)[1]
+        if self.model.base_gp.kern.measure is None:
```

Comment:
you mean as a property?

Reply:
maybe a property, yes. there is also a method two lines below it. idea is to just rely on model interface, instead of relying on the fact that model contains base_gp which contains kernel which contains measure. makes sense?

Reply:
I see, OK, I'll change it.